Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,779 @@
+    var fakeTableHeadContainer = scrollableContainer.parent().children('div.bocListFakeTableHead').first();
+    var fakeTableHead = fakeTableHeadContainer.find('thead');
+    var fakeTableHeadRow = fakeTableHead.children().eq(0);
+    var fakeTableHeadRowChildren = fakeTableHeadRow.children();
+  
+    // store cell widths in array
+    var realTableHeadCellWidths = new Array();
+    realTableHeadRowChildren.each(function (this: HTMLElement, index)
+    {
+      realTableHeadCellWidths[index] = $(this).width();
+    });
+  
+    // apply widths to fake header
+    fakeTableHeadRowChildren.width(function (index, itemWidth)
+    {
+      var width = realTableHeadCellWidths[index];
+      return width;
+    });
+  
+    var fakeTableHeadWidthContainer = fakeTableHeadContainer.children ('div').first();
+    fakeTableHeadWidthContainer.width(realTableWidth);
+    var fakeTableHeadContainerHeight = fakeTableHeadContainer.height();
+    scrollableContainer.css({ top: fakeTableHeadContainerHeight});
+    realTable.css({ 'margin-top': fakeTableHeadContainerHeight * -1 });
+  
+    fakeTableHeadContainer.show();
+  }
+
+  public static FixHeaderPosition(tableContainer: JQuery, scrollableContainerJQuery: JQuery): void
+  {
+    var scrollableContainer = scrollableContainerJQuery[0];
+    var fakeTableHeadContainer = tableContainer.children('div.bocListFakeTableHead').first()[0];
+    var scrollLeft = scrollableContainer.scrollLeft;
+    var previousScrollLeft = scrollableContainerJQuery.data("bocListPreviousScrollLeft");
+    var fakeTableHeadScrollLeft = fakeTableHeadContainer.scrollLeft;
+
+    var hasScrollMoveFromScrollbar = previousScrollLeft !== scrollLeft;
+    var hasScrollMoveFromColumnHeader = previousScrollLeft !== fakeTableHeadScrollLeft;
+    if (!hasScrollMoveFromScrollbar && !hasScrollMoveFromColumnHeader)
+      return;
+
+    scrollLeft = hasScrollMoveFromColumnHeader ? fakeTableHeadScrollLeft : scrollLeft;
+    scrollableContainerJQuery.data("bocListPreviousScrollLeft", scrollLeft);
+
+    if (hasScrollMoveFromColumnHeader)
+      scrollableContainer.scrollLeft = scrollLeft;
+
+    if (hasScrollMoveFromScrollbar)
+      fakeTableHeadContainer.scrollLeft = scrollLeft;
+  }
+
+  public static InitializeNavigationBlock (pageNumberField: JQuery, pageIndexField: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsObject ('pageNumberField', pageNumberField);
+    ArgumentUtility.CheckNotNullAndTypeIsObject('pageIndexField', pageIndexField);
+  
+    pageNumberField.bind('change', function () {
+  
+      var pageNumber = parseInt(pageNumberField.val(), 10);
+      if (isNaN (pageNumber) || !TypeUtility.IsInteger (pageNumber))
+      {
+        if (pageNumberField.val().length > 0)
+          setTimeout(function () { pageNumberField.focus(); }, 0);
+  
+        pageNumberField.val(String(parseInt(pageIndexField.val(), 10) + 1));
+        return false;
+      }
+      else
+      {
+        pageIndexField.val (String(pageNumber - 1));
```

Comment:
I actually do not know if `String(...)` works correctly because it is a constructor and should be used with `new`. I would just concat with an empty string.
```suggestion
        pageNumberField.val("" + (parseInt(pageIndexField.val(), 10) + 1));
        return false;
      }
      else
      {
        pageIndexField.val ("" + (pageNumber - 1));
```

Reply:
Is that really the TypeScript way to do it? To me it looks like a dumb JavaScript hack 😂

Reply:
TypeScript is a superset of JavaScript so dumb hacks also work in TypeScript ;)

Converting to a string like this is also common in C#. You could use `.toString()` or a template string as alternative options.